src/main/scala-2.12/kamon/scala/instrumentation/FutureInstrumentation.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2014 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.scala.instrumentation
+
+import kamon.trace.{TraceContextAware, Tracer}
+import org.aspectj.lang.ProceedingJoinPoint
+import org.aspectj.lang.annotation._
+
+@Aspect
+class FutureInstrumentation {
```

Comment:
Hmm, my style is to error on the side of resolving warnings rather than hiding them.
Yes, in this case, it may be safe to resolve by ignoring the message produced, but it would also hide future warnings.  

If you guys would prefer that the Xlint option be used, I'm happy to adjust this PR to reflect that change.